src/oop/static_modifier/medium/q4/MainTest.java

```diff
@@ -0,0 +1,141 @@
+                String.format(modificationErrorMessage, className));
+        assertTrue(testClass.hasModifier(secondIntName, "final"),
+                String.format(modificationErrorMessage, className));
+        assertTrue(testClass.hasModifier(arrayName, "private"),
+                String.format(modificationErrorMessage, className));
+        for (int i = 0; i < 10; i++) {
+            double[] test = generateRandomArray(0.5, 20.0, ((int) (Math.random() * 20)));
+            Object[][] arguments = {{test, double[].class}};
+            Object classInstance = testClass.createInstance(arguments);
+            _assertArrayEquals(test, testClass.getFieldValue(classInstance, arrayName),
+                    String.format(modificationErrorMessage, className));
+            _assertEquals(initialValue, testClass.getFieldValue(null, firstDoubleName),
+                    String.format(modificationErrorMessage, className));
+            _assertEquals(initialValue, testClass.getFieldValue(null, secondDoubleName),
+                    String.format(modificationErrorMessage, className));
+            _assertEquals(typeValue1, testClass.getFieldValue(null, firstIntName),
+                    String.format(modificationErrorMessage, className));
+            _assertEquals(typeValue2, testClass.getFieldValue(null, secondIntName),
+                    String.format(modificationErrorMessage, className));
+
+        }
+    }
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of(new double[]{initialValue, initialValue, initialValue, initialValue, initialValue}),
+                Arguments.of(new double[]{initialValue + 1, initialValue + 1, initialValue + 1, initialValue + 1, initialValue + 1}),
+                Arguments.of(new double[]{initialValue - 1, initialValue - 1, initialValue - 1, initialValue - 1, initialValue - 1}),
+                Arguments.of(generateRandomArray(0.5, 20.0, 10)),
+                Arguments.of(generateRandomArray(0.5, 20.0, 10)),
+                Arguments.of(generateRandomArray(0.5, 20.0, 10))
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void correctMixFunMethod(double[] guide) throws Throwable {
+        Object[][] arguments = {
+                {guide, double[].class}
+        };
+        assertTrue(testClass.hasMethod(methodName, null),
+                "Your " + className + " class does not have a " + methodName + " method");
+        assertTrue(testClass.hasMethod(methodName, null, Void.TYPE),
+                "Your " + methodName + " method should not return anything.");
+        assertTrue(testClass.hasMethod(methodName, null, Void.TYPE, new String[]{"public"}),
+                "Your " + methodName + " method does not have the correct visibility modifier.");
+        Object classInstance = testClass.createInstance(arguments);
+        double value1 = (double) testClass.getFieldValue(null, firstDoubleName);
```

Comment:
these values have already been defined on line 29, you should be able to access them here